access-common/src/main/java/edu/unc/lib/dl/ui/service/SolrQueryLayerService.java

```diff
@@ -1110,21 +1108,20 @@ public long getCollectionsCount() {
 	 * @return a map from format name to count
 	 */
 
-	public Map<String, Long> getFormatCounts() {
-
-		SolrQuery query;
-		QueryResponse response;
-
-		query = new SolrQuery();
-		query.setQuery("*:*");
-		query.setRows(0);
-		query.addFacetField("contentType");
-		query.setFacetLimit(-1);
-
+	public Map<String, Long> getFormatCounts(AccessGroupSet accessGroups) {
 		HashMap<String, Long> counts = new HashMap<String, Long>();
-
+		
 		try {
-			response = this.executeQuery(query);
+			StringBuilder queryBuilder = new StringBuilder("*:*");
```

Comment:
i wouldn't mind seeing a private function that encompasses the logic in lines 1115-1120, and their equivalents above. it could take a string param to pass to the StringBuilder constructor

Reply:
Trying to think of how that would be named, `makeSolrQueryWithAccessRestrictions`?  Hmm, I feel like `addAccessRestrictions` may already be an encapsulation of the common functionality here.  Often the two builders (StringBuilder and SolrQuery) would both need to be available to the method constructing a query, although query construction logic in these three cases is pretty simple.  I'm leaning towards leaving this as it is for now, but I would like to refactor this class significantly in the future since it doesn't have a real purpose, its basically a functional dumping ground of queries at present.

Reply:
ok, sounds reasonable